homeassistant/components/tplink/config_flow.py

```diff
@@ -270,14 +272,32 @@ async def async_step_user(
         if user_input is not None:
             if not (host := user_input[CONF_HOST]):
                 return await self.async_step_pick_device()
-            self._async_abort_entries_match({CONF_HOST: host})
+            host, _, port_str = host.partition(":")
```

Comment:
If host is an braketed ipv6 address this will split the address

Reply:
https://github.com/aio-libs/yarl/blob/dfc41dbfcd5a91283ab6c5c21d70bc4b1c377789/yarl/_parse.py#L123

Reply:
That's fixed thanks